cypress/pages/checkBox.page.js

```diff
@@ -6,12 +6,152 @@ const elementsCheckBox = new CheckBoxElements();
 
 export default class CheckBoxPage {
   clickOnCheckBoxButton() {
-    cy.get(elementsCheckBox.CHECK_BOX).click();
+    cy.get(elementsCheckBox.CHECK_BOX_BUTTON).click();
   }
   checkIfThereIsOnlyOneLabelWithTextHome() {
     cy.get(elementsCheckBox.ALL_LABELS).should("have.lengthOf", 1);
   }
-  expandAllLabels() {
-    cy.get(elementsCheckBox.ALL_LABELS).contains("Home").find(button).click();
+  expandOrCollapseLabel(labelName) {
+    cy.get(elementsCheckBox.ALL_LABELS)
+      .contains(labelName)
+      .parent("label")
+      .siblings("button")
+      .click();
+  }
+
+  checkInMoreLabelsAndAssert(moreLabels) {
+    for (var i = 0; i < moreLabels.length; i++) {
+      var label = moreLabels[i];
+      cy.log(label);
+      this.checkInSomeElementAndAssert(label);
+    }
+  }
+
+  checkInSomeElementAndAssert(elementName) {
+    cy.get("body").then(($body) => {
+      if ($body.find("span:contains(" + elementName + ")").length == 0) {
+        cy.get(elementsCheckBox.EXPAND_BUTTON).first().click();
+        this.checkInSomeElementAndAssert(elementName);
+      } else {
+        cy.get(elementsCheckBox.ALL_LABELS)
+          .contains(elementName)
+          .parent("label")
+          .children("input")
+          .invoke("show")
+          .check();
+      }
+      cy.get(elementsCheckBox.ALL_LABELS)
```

Comment:
Definitely that is better solytion to put it in else block, but much better is to only put .should('be.checked') after check() on 41. line.

Reply:
However, that code after the else block wouldn't be performed multiple times because the interpreter repeats the if block until it finds an element, and than goes to else and reaches that part of the code only one time. But in terms of logic and style your proposal is good and I agree.